docs/tutorial/gotchas.md

```diff
@@ -1,5 +1,7 @@
 ## Common Gotchas
 
+In the following, we summarize some common tips, tricks and gotchas to run smoothly Apache Spark on kubernetes, even in small local environments
+
 * For spark-submit to work correctly, make sure DNS is enabled.
```

Comment:
why put this info in a separate document? Put it as a "caution" note in the text along with the guide for using spark-submit, and/or together with the setup microk8s section. Maybe even put it as a prerequisite - working DNS in the cluster and outside.

Reply:
It felt to me natural to put this as the last part of the tutorial (see the index markdown, that would reflect how things may can be parsed from discourse)